ministryofjustice/correspondence_tool_staff
# Ct 3703 address lookup selector

UPDATE: PR now contains code that filters what the address lookup is able to look for to prison and probation on the subject details page.

## Description
adds feature to allow users to search and select addresses from a popup modal on the subject details page.

Note this uses a rails unobtrusive JS pattern as described here:
https://medium.com/backticks-tildes/unobtrusive-javascript-in-rails-c37fc757d8b1 

It means we use a smaller number manually set javascript events and the new modal forms are stored as partials rather than inline as JS strings.



 

## Self-review checklist


### Screenshots

After:

<img width="591" alt="Screenshot 2021-08-05 at 14 13 50" src="https://user-images.githubusercontent.com/13377553/128355858-f6acfd08-0b19-4859-9c46-847e9b417b75.png">

Click Find an address:
<img width="642" alt="Screenshot 2021-08-05 at 14 14 43" src="https://user-images.githubusercontent.com/13377553/128355979-61ded93d-3736-42f2-91c4-8b2d0dee3d16.png">

Search:
<img width="626" alt="Screenshot 2021-08-05 at 14 15 26" src="https://user-images.githubusercontent.com/13377553/128356082-77a369b8-6b64-4f18-bc78-b2c8b097753f.png">

Select
<img width="622" alt="Screenshot 2021-08-05 at 14 16 31" src="https://user-images.githubusercontent.com/13377553/128356228-e5918b17-6c37-48c5-ac89-66563e6840bf.png">


### Related JIRA tickets
[CT-3703](https://dsdmoj.atlassian.net/browse/CT-3703)

### Manual testing instructions
- make sure you have added at least one address on the Addresses page
- start a sar
- at the address field click 'Find an address'
- enter the name or a partial part of the name you used for your address
- select the address returned by clicking relevant button
- it should now be in the pasted into the address field.
